tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -725,8 +758,30 @@ class ScopedFilterDescriptor {
     const auto& spatial_dims = filter_descriptor.input_filter_dims();
     std::copy(spatial_dims.begin(), spatial_dims.end(), dims.begin() + 2);
 
```

Comment:
Could you implement the below in a dedicated member function to mimic `BatchDescriptor::full_strides(layout)`. I believe, with that, you can reuse the `ReorderDims()` function. This way, both tensor descriptor and filter descriptor will rely on one set of mechanism to calculate strides. `ReorderDims()` Seems to implement the exact way how MIOpen re-arrange strides.

Reply:
I had thought about going that way initially, but then chose this instead because
* this was easier
* the other options required putting the change in `dnn.[h,cc]` files, which are not ROCm specific, which make the change more suspectible to G scrutiny when upstreaming.

Now that we have one more vote for that route, let me code it up :)

Reply:
Thanks Deven! I can confirm that this branch look good, comment looks good. In case you need to do debugging any time, this is a good candidate to visualize how strides are calculated according to dims.

> the other options required putting the change in dnn.[h,cc] files, which are not ROCm specific
I agree, Cuda API may somehow do not require a layout related update on the filter side. I'd leave the details to you but you should be able to make `ReorderDims()` available in `rocm_dnn.cc` and only add this logic on our descriptor.

Reply:
done